api/src/main/java/org/pmiops/workbench/api/CohortReviewController.java

```diff
@@ -52,8 +168,31 @@
      * @return
      */
     @Override
-    public ResponseEntity<ParticipantCohortStatusListResponse>
-    getParticipantCohortStatuses(Long cohortId, Long cdrVersionId, Integer page, Integer limit, String order, String column) {
+    public ResponseEntity<org.pmiops.workbench.model.CohortReview>
+    getParticipantCohortStatuses(Long workspaceId, Long cohortId, Long cdrVersionId, Integer page, Integer limit, String order, String column) {
+
+        CohortReview cohortReview = cohortReviewDao.findCohortReviewByCohortIdAndCdrVersionId(cohortId, cdrVersionId);
+
+        if (cohortReview == null) {
+            CohortDefinition definition = cohortDao.findCohortByCohortIdAndWorkspaceId(cohortId, workspaceId);
+            if (definition == null) {
+                throw new BadRequestException("Invalid Request: No Cohort definition matching cohortId: "
+                        + cohortId + ", workspaceId: " + workspaceId);
+            }
+            SearchRequest request = new Gson().fromJson(definition.getCriteria(), SearchRequest.class);
+            QueryResult result = bigQueryService.executeQuery(
+                    bigQueryService.filterBigQueryConfig(participantCounter.buildParticipantCounterQuery(request)));
+            Map<String, Integer> rm = bigQueryService.getResultMapper(result);
+            List<FieldValue> row = result.iterateAll().iterator().next();
+            long cohortCount = bigQueryService.getLong(row, rm.get("count"));
+
+            cohortReview = new CohortReview();
+            cohortReview.setCohortId(cohortId);
+            cohortReview.setCdrVersionId(cdrVersionId);
+            cohortReview.matchedParticipantCount(cohortCount);
+            cohortReview.setCreationTime(new Timestamp(System.currentTimeMillis()));
```

Comment:
Do you need to set the review status here?

Reply:
Done.